### ISSUE TITLE
Using gh-ost over an ssh tunnel on non-standard mysql port? [question]

### ISSUE THREAD

**Issue by jaygooby** [github/gh-ost#631]
I'm just starting to experiment with using gh-ost, and I know I'm using it in a slightly awkward way, because I'm running it from my local machine over an ssh tunnel to an RDS instance I've started, so that I can experiment...

I start the tunnel like this:

```
ssh -f -o ExitOnForwardFailure=yes -L 43992:rds-test.example.com:3306 me@example.com sleep 10
```

Where `43992` is a random unused port on my local machine, `rds-test.example.com` is my AWS RDS instance, and `me@example.com` is a user on an AWS machine in the same VPS as `rds-test.example.com`

If I connect a regular mysql client like this:

```
mysql -u rdstest -h 127.0.0.1 -P 43992 -p
```

Then I can connect via the tunnel fine, but when I try the same method using gh-ost, I get:

```
FATAL Unexpected database port reported: 3306
```

which I guess is due to https://github.com/github/gh-ost/search?utf8=%E2%9C%93&q=Unexpected+database+port+reported&type= and the server itself running on `3306`, but the gh-ost client being tunneled through `43992`.

It's not a show-stopper for me, because I can stop my local mysql instance and run the tunnel over local port `3306`. Any suggestions other than doing this?

**Comment by shlomi-noach** [comment]
As a hackish solution, try passing `--aliyun-rds`; in Aliyun RDS the server refuses to report the `@@port` and the port check is silenced.

**Comment by jaygooby** [comment]
Great, that did it :) 

I was religiously searching the `--help` options for any other mention of "port" so completely didn't clock the `-aliyun-rds` or its real purpose ;)

Would you like a PR for some better help text for that option? Something like:

```
-aliyun-rds
        set to 'true' when you execute on Aliyun RDS. Prevents gh-ost from complaining ('Unexpected database port reported') when the user-supplied `--port` is different to the RDS mysql port
```

**Comment by shlomi-noach** [comment]
Thank you, no, actually I'd like to avoid that; tomorrow `aliyun` may have a different requirement that doesn't have to do something with the `--port`. `-aliyun-rds` is there to support Aliyun RDS, not to necessarily support port forwarding.

We could consider a new option, like `-force-port`, but at this time I'm not sure the incentive is strong.

**Comment by shlomi-noach** [comment]
Please see #632

**Comment by mahemoff** [comment]
I have the same issue with Docker. It runs inside the container using port 3306 and exposed as 3307, which leads to "FATAL Unexpected database port reported: 3306".

Seeing this with gh-ost 1.0.47 when using --port=3307. I guess the above

**Comment by shlomi-noach** [comment]
@mahemoff and did `-aliyun-rds` or `-gcp` work for you?

**Comment by mahemoff** [comment]
Yes, those do work. I just thought it had been fixed in 1.0.47, but I can use those for now.
